src/warc2zim/url_rewriting.py

```diff
@@ -135,17 +135,23 @@ def normalize(url: str | bytes) -> str:
     return path
 
 
+def remove_anchor(url: str) -> str:
```

Comment:
```suggestion
def get_with_anchor(url: str) -> str:
```

Since you're returning a modified version and not modifying in-place. I like _fragment_ better than anchor but both are understandable